### PR TITLE
Fixed Wiki Grammar

### PR DESCRIPTION
**Pull request by WalkerKnapp**
Fixes various grammar errors in the wiki. This is split into many different commits since the editing was done in Github's web interface. If you want me to squash them, I can.

### CODE REVIEW

File: _wiki/essential-knowledge/working-with-futures.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ keywords:
 
 > This tutorials assumes, you are familiar with lambda expressions. Take a look at the [Lambda Introduction](/wiki/essential-knowledge/lambda-introduction) first, if you are not!
 
-As Javacord is heavily multithreaded, you have to understand the concept of [Futures](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html) in general, as well as their most common implementation, the [CompletableFuture](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html). This little introduction should give you a quick overview about the basics you need to know in order to work with Futures.
+As Javacord is heavily multithreaded, you have to understand the concept of [Futures](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html) in general, as well as their most common implementation, the [CompletableFuture](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html). This little introduction should give you a quick overview of the basics you need to know in order to work with Futures.
```
Comment (Bastian) [inline _wiki/essential-knowledge/working-with-futures.md:14]: Is this really correct?
https://ell.stackexchange.com/questions/11583/provides-information-on-of-or-about-somthing

File: _wiki/essential-knowledge/working-with-futures.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ System.out.println((System.currentTimeMillis() - currentTime) + " ms");
 
 ## join()
 
-This method blocks the current thread, until the method finished and returns its result or throwing a `CompletionException` if anything failed. You usually should avoid `join()` for larger bots or methods which will be called frequently.
```
Comment (Bastian) [inline _wiki/essential-knowledge/working-with-futures.md:44]: Wow, I managed to mix 3 different tenses ^^

File: _wiki/essential-knowledge/working-with-futures.md [hunk 3]
```diff
@@ -58,15 +58,15 @@ Message message = channel.sendMessage("First!").join();
 // join() ensures, that an exception is thrown in case something went wrong
 message.addReaction("👍").join();
 ```
-`join()` can basically be used every time if you don't care that much about your application's performance. So if you're only developing a small bot for your own server and your current goal is to just get it working, it is probably easier to always use `join()` to get the result of a method. You'll usually not run into problems if you don't overuse this method.
+`join()` can basically be used every time if you don't care that much about your application's performance. So if you're only developing a small bot for your own server and your current goal is to just get it working, it is probably easier to always use `join()` to get the result of a method. You'll won't usually run into problems if you don't overuse this method.
```
Comment (Bastian) [inline _wiki/essential-knowledge/working-with-futures.md]: Isn't `You'll won't` wrong? I guess you wanted to write `You won't`?

**Comment by Bastian** [comment]
Thank you 👍 

```
If you want me to squash them, I can.
```
Would be nice :)

File: _wiki/essential-knowledge/working-with-futures.md (hunk 1 above)
Comment (WalkerKnapp) [reply _wiki/essential-knowledge/working-with-futures.md:14]: I believe so, although after looking over it, both are correct. The key difference is that in that link `provides information` is essentially a verb, which `on`, `of`, and `about` operate on, but in the wiki `a quick overview` is essentially a noun. `an overview of` taken out of context is correct, as far as I can tell, and is used more often in regular language than `an overview about`. So, it's up to you which you want to use.

File: _wiki/essential-knowledge/working-with-futures.md (hunk 3 above)
Comment (WalkerKnapp) [reply _wiki/essential-knowledge/working-with-futures.md]: Yep, thanks for catching that.

File: _wiki/advanced-tutorials/entity-cache.md [hunk 4]
```diff
@@ -80,4 +80,4 @@ api.addReactionAddListener(event -> {
 
 Some examples when cached entities get invalid:
```
Comment (Mysterypotatoguy) [inline _wiki/advanced-tutorials/entity-cache.md]: `Some examples of when cached entities become invalid` or `Some examples of when cached entities are invalidated` ?

Comment (WalkerKnapp) [reply _wiki/advanced-tutorials/entity-cache.md]: 2nd in my opinion. Thanks for the catch 👍

File: _wiki/essential-knowledge/working-with-futures.md (hunk 1 above)
Comment (Bastian) [reply _wiki/essential-knowledge/working-with-futures.md:14]: Let's go with `of` then, if you think that's more common.

**Pull request merged**